### PR TITLE
Support for shoryuken background job processor and logging to stdout

### PR DESCRIPTION
**Pull request by stemps**
👋 Hi Lieferies! 😀

I'm using `loggery-gem` on my current project and have added 2 changes:

1. support for the [shoryuken](https://github.com/phstc/shoryuken) background processor (sidekiq-clone for AWS SQS)
2. support for different LogstashLogger log-types to support logging to stdout, like us docker-kids do

Let me know if you're ok with this and I'm also happy to do a version bump and publish the gem. I think I'm still the maintainer for that anyway ;-)

Simon

### CODE REVIEW

File: lib/loggery/metadata/middleware/shoryuken.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+module Loggery
+  module Metadata
+    module Middleware
+      class Shoryuken
+        include Loggery::Util
+
+        # Clients can provide their own error handler
+        class << self
+          attr_accessor(:error_handler) { ->(e) { Rails.logger.error(e) } }
```
Comment (sliminas) [inline lib/loggery/metadata/middleware/shoryuken.rb:13]: Maybe the comment should go to the next line?
```suggestion
        class << self
          # Clients can provide their own error handler
          attr_accessor(:error_handler) { ->(e) { Rails.logger.error(e) } }
```

File: spec/loggery/metadata/middleware/shoryuken_spec.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Loggery::Metadata::Middleware::Shoryuken.error_handler = nil
+      end
+
+      it "logs and re-raises the exception with correct metadata" do
+        allow(Rails.logger).to receive(:info)
+          .with(event_type: :shoryuken_job_started, message: anything, execution_delay: anything)
+          .and_call_original
+        allow(Rails.logger).to receive(:info)
+          .with(event_type: :shoryuken_job_finished, message: anything, duration: anything)
+          .and_call_original
```
Comment (sliminas) [inline spec/loggery/metadata/middleware/shoryuken_spec.rb:57]: Shouldn't those be expectations?
```suggestion
        expect(Rails.logger).to receive(:info)
          .with(event_type: :shoryuken_job_started, message: anything, execution_delay: anything)
          .and_call_original
        expect(Rails.logger).to receive(:info)
          .with(event_type: :shoryuken_job_finished, message: anything, duration: anything)
          .and_call_original
```
If not, why is it necessary at all?

**Review by sliminas — COMMENTED** [review]
Looks good to me.

File: spec/loggery/metadata/middleware/shoryuken_spec.rb (hunk 2 above)
Comment (stemps) [reply spec/loggery/metadata/middleware/shoryuken_spec.rb]: turns out... it isn't 😀 ...removed it

**Comment by stemps** [comment]
ping 😉

**Review by sliminas — APPROVED** [review]
Looks good to me. :+1:

**Comment by sliminas** [comment]
@adamniedzielski What do you think?

**Review by adamniedzielski — APPROVED** [review]
Let's give it a try! 🌷 🍺

**Pull request merged**